src/plugins/data/public/ui/dataset_navigator/dataset_navigator.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { EuiButtonEmpty, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import { SavedObjectsClientContract, SimpleSavedObject } from 'opensearch-dashboards/public';
+import { map, scan } from 'rxjs/operators';
+import { ISearchStart } from '../../search/types';
+import { IIndexPattern } from '../..';
+import { getUiService, getIndexPatterns, getSearchService } from '../../services';
+
+const getClusters = async (savedObjectsClient: SavedObjectsClientContract) => {
+  return await savedObjectsClient.find({
+    type: 'data-source',
+    perPage: 10000,
+  });
+};
+
+export const searchResponseToArray = (showAllIndices: boolean) => (response) => {
```

Comment:
nit. "all" indices might be ambiguous, would it be better with `includeHiddenIndices`?

and there's the `IOpenSearchDashboardsSearchResponse` type although raw response is still `any` by default
```suggestion
export const searchResponseToArray = (showAllIndices: boolean) => (response: IOpenSearchDashboardsSearchResponse) => {
```